.github/workflows/build.yaml

```diff
@@ -12,7 +12,6 @@ jobs:
         rust:
           - nightly
           - stable
-          - 1.34.2
```

Comment:
Rust 1.34.2 is the newest version supported in Debian stable: https://packages.debian.org/buster/rustc. I want the crate to be compilable there, hence I'd prefer not to drop support for it.

To me, mechanically replacing `mem::uninitialized()` with `mem::MaybeUninit::uninit().assume_init()` does not seem worth bumping the minimum supported Rust version. If you want to achieve a warning-free build, we can create a `common::mem::uninitialized` function that just calls `mem::uninitialized()` but has an `#[allow(deprecated)]` annotation to suppress the warning.

Reply:
Do we want to support rustc from Debian stable? I think given that rust
a) is being actively developed b) has it's own tool for installing the latest version - rustup c) lists this tool as the only installation method on their website - we could consider stepping away from supporting the packaged version.

But I'm also ok with keeping 1.34.2 since that was admittedly the only place where we've encountered a problem yet.

Btw, what do you think about removing those uninitialized buffers completely (making them initialized)? According to [rust doc](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html), we get undefined behaviour even with i32:

```
use std::mem::{self, MaybeUninit};

let x: i32 = unsafe { mem::uninitialized() }; // undefined behavior! ⚠️
// The equivalent code with `MaybeUninit<i32>`:
let x: i32 = unsafe { MaybeUninit::uninit().assume_init() }; // undefined behavior! ⚠️
```
I suspect that the overhead from initialising buffers in our case would be negligible.

Reply:
For reference: We use Debian stable on all DDNet servers. Seems like a popular enough distribution to support.

Reply:
@def- we could use rustup to install rustc. It's listed as the official installation method so there shouldn't be many problems with it. I've just switched to it on my ubuntu desktop to install nightly toolchain for https://github.com/google/rerast. It's capable of keeping arbitrary number of toolchains installed for different purposes.

But again I'm also ok with keeping 1.34.2 supported for now since we only encountered one issue with it.

Reply:
> Do we want to support rustc from Debian stable? I think given that rust
> a) is being actively developed b) has it's own tool for installing the latest version - rustup c) lists this tool as the only installation method on their website - we could consider stepping away from supporting the packaged version.

Yes, I want to support Rust from Debian stable. (Rust isn't special. All languages I know are actively developed, be it C++ with gcc or clang, Python with CPython or Rust. I want to be able to use Rust without using some dedicated Rust installer, I prefer using package sources.)